config/append.go

```diff
@@ -64,7 +64,11 @@ func appendStruct(vOld, vNew reflect.Value) reflect.Value {
 		case reflect.Slice:
 			vfRes.Set(reflect.AppendSlice(vfOld, vfNew))
 		default:
-			vfRes.Set(vfNew)
+			if vfNew.Kind() != reflect.Ptr || !vfNew.IsNil() {
```

Comment:
nit: I think it would be more clear to flip the if clauses and say `if vfNew.Kind() == reflect.Ptr && vfNew.IsNil()`

Reply:
Agreed. Updated.